modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/Netty4Plugin.java

```diff
@@ -115,7 +120,12 @@ public Map<String, Supplier<HttpServerTransport>> getHttpTransports(
                 tracer,
                 TLSConfig.noTLS(),
                 null
-            )
+            ) {
+                @Override
+                protected void populateRequestThreadContext(RestRequest restRequest, ThreadContext threadContext) {
+                    dispatcherContext.accept(restRequest.getHttpRequest(), threadContext);
+                }
+            }
```

Comment:
This is the Http handler when security is disabled.

In this case, just use the consumer that populates the request-wise thread context in the default way. That consumer is furnished by the `RestController`. This is unfortunate, it would've been simpler if the `Netty4HttpServerTransport` were responsilbe to populate the thread context before delegating to the REST dispatcher.

Reply:
I think this is one area where we can discuss improvements.

Reply:
> But the Security pugin can only intercept the http server transport and not the rest controller, and it's the rest controller that contains the information necessary to populate the context (the list of HTTP headers to copy in the thread context).

For more details see the `Security` class, eg https://github.com/elastic/elasticsearch/pull/93656#discussion_r1111783159 .

Security needs to use the baseline thread context supplier, that's been built by the `RestController`, in the authenticating netty channel handler, which is constructed together with the http server transport by the Security plugin.